src/biokbase/narrative/jobs/job.py

```diff
@@ -322,21 +303,47 @@ def state(self, force_refresh=False):
         if self.was_terminal and not force_refresh:
             state = copy.deepcopy(self._acc_state)
         else:
-            try:
-                state = clients.get("execution_engine2").check_job(
-                    {"job_id": self.job_id, "exclude_fields": EXCLUDED_JOB_STATE_FIELDS}
-                )
-                state = self.update_state(state)
-
-            except Exception as e:
-                raise Exception(f"Unable to fetch info for job {self.job_id} - {e}")
+            state = self.query_ee2_state(self.job_id, False)
+            state = self.update_state(state)
 
         for field in EXCLUDED_JOB_STATE_FIELDS:
             if field in state and field != "job_input":
                 del state[field]
 
         return state
 
+    @staticmethod
```

Comment:
Could refactor so `query_ee2_state` calls `query_ee2_states` but the existing mocking in tests is prohibitive

Reply:
Also the exception catching was dropped. Not sure if it did anything besides provide a less scary (albeit less useful!) error message

Reply:
Putting in exception catching here means not necessarily having it wherever else you call `query_ee2_state`. But it probably makes sense to either let the errors percolate up or catch them at some other layer.

Reply:
Okay good. It was really before just catching and rethrowing the error but losing the stack trace in the process. I couldn't think of a reason why I needed to keep it